packages/angular_devkit/schematics/src/tree/host-tree_spec.ts

```diff
@@ -26,6 +26,23 @@ describe('HostTree', () => {
     });
   });
 
+  describe('readJSON', () => {
+    it('returns a JSON value when reading a file that exists', () => {
+      const tree = new HostTree();
+      tree.create('/textfile1', '{ "a": true, "b": "xyz" }');
+      tree.create('/textfile2', '123');
+      tree.create('/textfile3', 'null');
+      expect(tree.readJSON('/textfile1')).toEqual({ a: true, b: 'xyz' });
+      expect(tree.readJSON('/textfile2')).toEqual(123);
+      expect(tree.readJSON('/textfile3')).toBeNull();
```

Comment:
Nit: Should test JSON files with comments and trailing commas, since that's an explicit part of its contract.

Should also test a malformed JSON file and confirm the failure case.